### PR TITLE
Remove default classpath script for db-init-script

### PR DESCRIPTION
**Pull request by AnanaMJ**
And some other changes to fix some things encountered on the way:
* Changed the name of the database file for h2 to reflect the name of the schema in the configuration file (instead of having `housekeeping.mv.db` for any schema name)
* Added the configuration directory to the classpath by default for vacuum tool instead if having to specify `logging` in the vacuum tool config.

Corresponding Circus Train PR coming soon.

### CODE REVIEW

File: housekeeping-vacuum-tool/README.md [hunk 1]
```diff
@@ -60,11 +60,11 @@ The `dry-run` option allows you to observe the status of paths on the file syste
 |`tables.table-name`|Yes| The Hive table name for the table the vacuum tool will interrogate.|
 |`housekeeping.schema-name`|No|The schema name that is used in your housekeeping instance. Defaults to `housekeeping`.|
 |`housekeeping.h2.home`|No|The path to your H2 filesystem database specified as a full path. Defaults to `$HOME/data`.|
```
Comment (AnanaMJ) [inline housekeeping-vacuum-tool/README.md]: I wasn't sure about this, but I got the feeling that this row should be

|`housekeeping.h2.database`|No|The path to your H2 filesystem database specified as a full path. Defaults to `${instance.home}/data/${instance.name}/${housekeeping.schema-name}` (where `instance.home`, `instance.name` and `housekeeping.schema-name` can be configured separately for more fine-grained control).|
| :-- | :-- | :-- |

File: housekeeping/src/main/java/com/hotels/housekeeping/HousekeepingConfiguration.java [hunk 2]
```diff
@@ -76,8 +76,9 @@ public void postConstruct() {
         .put("spring.datasource.max-wait", 10000)
         .put("spring.datasource.max-active", 2)
         .put("spring.datasource.test-on-borrow", true)
-        .put("spring.datasource.schema", "${housekeeping.db-init-script:classpath:/schema.sql}")
-        .put("housekeeping.h2.database", "${instance.home}/data/${instance.name}/housekeeping")
+        .put("spring.datasource.schema", "${housekeeping.db-init-script: }")
+        .put("housekeeping.h2.database",
+            "${instance.home}/data/${instance.name}/${housekeeping.schema-name:housekeeping}")
```
Comment (AnanaMJ) [inline housekeeping/src/main/java/com/hotels/housekeeping/HousekeepingConfiguration.java]: I changed this line so when H2 is used, a database with the file with the same name as the schema name is created. I tested it for schema `custom_schema`, and instead of getting `/data/housekeeping.mv.db` (what I got with the latest released version of HK), I got `/data/custom_schema.mv.db`.

File: pom.xml [hunk 3]
```diff
@@ -80,4 +82,17 @@
       <scope>test</scope>
     </dependency>
   </dependencies>
+  
+  <reporting>
```
Comment (patduin) [inline pom.xml]: why is this added?

File: pom.xml [hunk 4]
```diff
@@ -26,11 +26,13 @@
   </scm>
 
   <properties>
+	<jdk.version>1.7</jdk.version>
     <shade.prefix>${project.groupId}.shaded</shade.prefix>
     <hadoop.version>2.7.1</hadoop.version>
     <hive.version>2.3.2</hive.version>
     <spring-platform.version>2.0.3.RELEASE</spring-platform.version>
     <maven-assembly-plugin.version>3.1.0</maven-assembly-plugin.version>
+    <maven-assembly-plugin.version>3.1.0</maven-assembly-plugin.version>
```
Comment (patduin) [inline pom.xml]: duplicate?

File: housekeeping-vacuum-tool/src/main/scripts/common.sh [hunk 5]
```diff
@@ -33,6 +33,12 @@ else
   export HOUSEKEEPING_CLASSPATH=$HOUSEKEEPING_CLASSPATH:$HOUSEKEEPING_LIBS
 fi
 
+HOUSEKEEPING_CONF=$HOUSEKEEPING_TOOL_HOME/conf/
+
+if [[ ! -z ${HOUSEKEEPING_CONF-} ]]; then
```
Comment (patduin) [inline housekeeping-vacuum-tool/src/main/scripts/common.sh]: this checks if  `HOUSEKEEPNG_CONF` has a value right but won't it always because we hardcode `/conf/` to it?

File: pom.xml (hunk 4 above)
Comment (AnanaMJ) [reply pom.xml]: Oh my, I'll remove this right away!

File: pom.xml (hunk 3 above)
Comment (AnanaMJ) [reply pom.xml]: Because `hotels-oss-parent` v4 removed `findbugs-maven-plugin` from reporting

File: housekeeping-vacuum-tool/src/main/scripts/common.sh (hunk 5 above)
Comment (AnanaMJ) [reply housekeeping-vacuum-tool/src/main/scripts/common.sh]: Actually, yeah, you're right; I can export `HOUSEKEEPING_CLASSPATH` outside the `if` statement.

File: housekeeping-vacuum-tool/README.md [hunk 6]
```diff
@@ -60,11 +60,11 @@ The `dry-run` option allows you to observe the status of paths on the file syste
...
-|`housekeeping.datasource.driver-class-name` |No|The fully qualified class name of your database driver. Defaults to the H2 driver if not configured.|
-|`housekeeping.datasource.url` |No| JDBC URL for your database. Defaults to H2 filesystem database if not specified. |
-|`housekeeping.datasource.username` |No| Username for your database.|
-|`housekeeping.datasource.password` |No| Password for your database.|
-|`housekeeping.db-init-script`|No|A file containing a script to initialise your schema can be provided if it does not already exist. Defaults to `classpath:/schema.sql`.|
+|`housekeeping.data-source.driver-class-name` |No|The fully qualified class name of your database driver. Defaults to the H2 driver if not configured.|
```
Comment (patduin) [inline housekeeping-vacuum-tool/README.md]: did we change the name of the datasoure property? (to data-source)

File: housekeeping-vacuum-tool/README.md (hunk 1 above)
Comment (patduin) [reply housekeeping-vacuum-tool/README.md]: I think you are right from the code: `.put("housekeeping.h2.database", "${instance.home}/data/${instance.name}/housekeeping")`

File: housekeeping-vacuum-tool/README.md (hunk 6 above)
Comment (AnanaMJ) [reply housekeeping-vacuum-tool/README.md]: This was resolved, but for the next person reviewing this: 
I didn't change the code for the key name. I just changed the docs so the key name is the same as the one in `houskeeping-parent` docs :)

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,12 @@
 ## TBD
 ### Changed
 * Changed recursion logic to prevent unnecessary calls.
+* Changed the name of the database file for h2 to reflect the name of the schema in the configuration file (instead of having `housekeeping.mv.db` for any schema name).
```
Comment (abhimanyugupta07) [inline CHANGELOG.md]: I think Changed is already said by the heading. So, all the statements should start from just what is being changed.
eg. 

Changed
* the name of the database....

Comment (abhimanyugupta07) [reply CHANGELOG.md]: Same for all other statements that are changed.

File: pom.xml [hunk 8]
```diff
@@ -26,6 +26,7 @@
   </scm>
 
   <properties>
+	<jdk.version>1.7</jdk.version>
```
Comment (abhimanyugupta07) [inline pom.xml]: formatting ?

Comment (AnanaMJ) [reply pom.xml]: If you're saying that I need to replace the tab with spaces, you're right!

If you're asking if this is a formatting change, it isn't. `hotels-oss-parent` provides jdk 8 by default now :)

Comment (abhimanyugupta07) [reply pom.xml]: Yeah. I meant the first one.

File: housekeeping-vacuum-tool/src/main/scripts/common.sh [hunk 9]
```diff
@@ -33,6 +33,9 @@ else
   export HOUSEKEEPING_CLASSPATH=$HOUSEKEEPING_CLASSPATH:$HOUSEKEEPING_LIBS
 fi
 
+HOUSEKEEPING_CONF=$HOUSEKEEPING_TOOL_HOME/conf/
```
Comment (patduin) [inline housekeeping-vacuum-tool/src/main/scripts/common.sh]: Not sure which PR you want to add this but the change is also in the readme-update pr
